backend/src/main/java/io/dcbn/backend/passwordReset/controllers/PasswordController.java

```diff
@@ -0,0 +1,67 @@
+package io.dcbn.backend.passwordReset.controllers;
+
+import io.dcbn.backend.authentication.models.DcbnUser;
+import io.dcbn.backend.authentication.repositories.DcbnUserRepository;
+import io.dcbn.backend.passwordReset.models.MailType;
+import io.dcbn.backend.passwordReset.models.ResetRequest;
+import io.dcbn.backend.passwordReset.services.MailService;
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.JwtException;
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.security.Keys;
+import javax.validation.Valid;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.server.ResponseStatusException;
+
+@RestController
+public class PasswordController {
+
+  private final MailService mailService;
+  private final DcbnUserRepository userRepository;
+  private final PasswordEncoder passwordEncoder;
+
+  @Value("${jwt.secret}")
+  private String secret;
+
+  @Autowired
+  public PasswordController(MailService mailService,
+      DcbnUserRepository userRepository,
+      PasswordEncoder passwordEncoder) {
+    this.mailService = mailService;
+    this.userRepository = userRepository;
+    this.passwordEncoder = passwordEncoder;
+  }
+
+  @PostMapping("/request-password")
+  public void requestPassword(@RequestBody String email) {
+    mailService.sendMail(email, MailType.PASSWORD_RESET);
+  }
+
+  @PostMapping("/reset-password")
+  public void resetPassword(@Valid @RequestBody ResetRequest resetRequest) {
```

Comment:
Ist das mit dem ResetRequest nicht etwas übertrieben? Oder muss man das hier in einer Klasse auslagern?

Reply:
Wenn das Ganze im Request Body stehen soll, dann ja wir könnten theoretisch auch nur das neue Passwort im Body stehen haben, und den Token als URL-Parameter mitschicken.
```suggestion
  public void resetPassword(@RequestParam String token, @RequestBody String newPassword) {
```
So ungefähr.

Reply:
Ne kann ruhig beides im Body stehen, oder? Sehe da nicht so den vorteil das eine in der URL zu haben

Reply:
Ja ich denke auch das es so wie es ist besser ist.

Reply:
Ne meinte, dass einfach beides als separate Parameter mit @RequestBody übergeben werden. Weil du die Klasse ja sonst nirgends nutzt

Reply:
Es kann nur einen RequestBody geben. Deswegen hab ich die Parameter in einer Klasse zusammengefasst.